### ISSUE TITLE
100% CPU usage

### ISSUE THREAD

**Issue by mixmastamyk** [keithgg/puddletag#337]
Hi,

For some reason I get runaway cpu usage when I have more than 15 music files in a folder.  Doesn't seem dependent on the files because I can move different ones in and out, and the CPU usage starts on the 15th.

When I hit Ctrl+C, it spits this out:
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/puddletag-1.2.0-py2.7.egg/puddlestuff/tagmodel.py", line 1004, in rowCount
    def rowCount(self, index = QModelIndex()):
KeyboardInterrupt
```
Not sure if it is useful.

**Comment by keithgg** [comment]
Hi, please attach a screenshot of the app when it does that. Also post the version of the libaries you're 
using from the Help -> About puddletag page menu.

**Comment by mixmastamyk** [comment]
Ok, will do this weekend hopefully.

**Comment by mixmastamyk** [comment]
Hi, it looks the same as when it runs normally.  I will try to poke at it a bit more to see if I can find the determining factor.  Here is the about dialog:

![Uploading pud.jpg…]()

**Comment by keithgg** [comment]
Hey, any more luck with this?

**Comment by mixmastamyk** [comment]
Hi, have been busy for a few weeks, will work on my music this weekend.  

It is hard to identify why it happens exactly, but it almost always does when I'm using the program for a few minutes or more.  I have no idea what it is doing in the background, does it start background threads?

**Comment by keithgg** [comment]
Nope. At most you'll have two threads happening at once. Once for the GUI and another for whatever processing it's doing (loading files, saving, etc).

Can you run iotop on your system and check if there's a lot of IO happening as well?

**Comment by mixmastamyk** [comment]
Tried again and it did start looping again.  This time it started for no particular reason after window was open for a few minutes and I ignored it.  Not sure why it happens, perhaps it doesn't like the unicode chars in filenames?  Grasping at straws, I know.

Doesn't seem to handle uploads :-/.

![Uploading pud2.jpg…]()

**Comment by keithgg** [comment]
I can't see your screenshot. Try uploading again.

Which desktop enviroment? Distribution?

**Comment by mixmastamyk** [comment]
I use Ubuntu Mate.

![pud2](https://cloud.githubusercontent.com/assets/767039/24517353/d515547c-1532-11e7-914f-73d4a2238fba.png)

**Comment by keithgg** [comment]
I'm fairly sure it's a Mate/Gnome issue, but can't even take a stab at what it could be. I'll try and get a VM up and running when I get some time and see if I can reproduce.

**Comment by keithgg** [comment]
Someone reported a similar issue at https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1523389.html

Can you run puddletag with strace and see if you get the same output?

    strace /usr/bin/puddletag
